x-pack/plugins/security_solution/public/common/components/sourcerer/use_pick_index_patterns.tsx

```diff
@@ -155,11 +157,11 @@ export const usePickIndexPatterns = ({
   // when scope updates, check modified to set/remove alerts label
   useEffect(() => {
     onSetIsModified(
-      selectedOptions.map(({ label }) => label),
-      dataViewId
+      selectedPatterns.map((pattern) => pattern),
```

Comment:
`isModified` should only be set when the redux state changes, `selectedOptions` is local component state and `selectedPatterns` comes from redux